### PR TITLE
use node:test instead of tap

### PR DESCRIPTION
**Pull request by big-kahuna-burger**
#### Checklist

- [x] run `npm run test` and `npm run benchmark`
- [x] tests and/or benchmarks are included
- [x] documentation is changed or added
- [x] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

File: test/global-decompress.test.js [hunk 1]
```diff
@@ -18,8 +18,8 @@ function createPayload (compressor) {
   return payload
 }
 
-test('It should decompress the request payload :', async (t) => {
-  t.test('using brotli algorithm when `Content-Encoding` request header value is set to `br`', async (t) => {
+describe('It should decompress the request payload :', async () => {
+  test('using brotli algorithm when `Content-Encoding` request header value is set to `br`', async (t) => {
     t.plan(2)
```
Comment (Uzlopak) [inline test/global-decompress.test.js:23]: but node:test has no t.plan?

Comment (big-kahuna-burger) [reply test/global-decompress.test.js:23]: https://nodejs.org/api/test.html#contextplancount ?

Comment (big-kahuna-burger) [reply test/global-decompress.test.js:23]: TIL. You just need to use `t.assert.<someassertionFn>` to make it count.

**Review by mcollina — APPROVED** [review]
lgtm

**Comment by Uzlopak** [comment]
hmm. strange. t.plan is supported since 22.2 
How is it possible that node 20 passes?

I thought you need @mcollina/tspl

**Comment by big-kahuna-burger** [comment]
> hmm. strange. t.plan is supported since 22.2 How is it possible that node 20 passes?
> 
> I thought you need @mcollina/tspl

Backported since 20.15.0

See 20 docs here [context.plan(count)](https://nodejs.org/docs/latest-v20.x/api/test.html#contextplancount)

**Review by gurgunday — APPROVED** [review]
lgtm, thanks!

File: .c8rc [hunk 2]
```diff
@@ -0,0 +1,14 @@
+{
+  "all": true,
+  "exclude": [
+    "test/",
+    "coverage/",
+    "types/"
+  ],
+  "clean": true,
+  "check-coverage": true,
```
Comment (gurgunday) [inline .c8rc:9]: Do we need to set this to true and have the rest of the settings? Doesn’t c8 by default checks all these?

Comment (big-kahuna-burger) [reply .c8rc:9]: Setting this to `true` tells c8: "if below given threshold on any of the branches | lines | functions | statements" - exit process with code 1.
Setting to `false`, reports coverage anyway - but exit with 0.

Comment (big-kahuna-burger) [reply .c8rc:9]: See here:
Where it's decided to checkCoverage based on given args or rc files:
https://github.com/bcoe/c8/blob/ff146b4dde004c62651b57c33cedd8353c94c423/lib/commands/report.js#L43

And here is exitCode 1 when per-file or global thresholds fail:
https://github.com/bcoe/c8/blob/ff146b4dde004c62651b57c33cedd8353c94c423/lib/commands/check-coverage.js#L59

**Pull request merged**